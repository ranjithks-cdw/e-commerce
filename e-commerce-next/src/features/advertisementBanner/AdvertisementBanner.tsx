import Banner from './banner/Banner';
import './styles.css';
import { adBannerType } from "./types/adBannerTypes";

const AdvertisementBanner = async () => {
    const adData = getBannerData();

    const [bannerData] = await Promise.all([adData]);

    return (
        <Banner {...bannerData}/>
     );
};

const getBannerData = async () => {
    const genAdData = () => {
        const adData = {
            adName: "Title 1",
            adImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
            adRedirectUrl: "/products"
        };
        return adData;
    };
    
    const adData = genAdData();

    return adData;
}
 
export default AdvertisementBanner;