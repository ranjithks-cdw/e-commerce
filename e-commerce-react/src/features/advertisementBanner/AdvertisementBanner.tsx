import { useEffect, useState } from 'react';
import Banner from './components/banner/Banner';
import './styles.css';
import { adBannerType } from "./types/adBannerTypes";

const AdvertisementBanner = () => {
    const [adData, setAdData] = useState<adBannerType>();
    
    useEffect(() => {
        const genAdData = () => {
            const adData = {
                adName: "Title 1",
                adImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
                adRedirectUrl: "/products"
            };
            return adData;
        };

        const bannerData = genAdData();
        setAdData(bannerData);

    },[]);

    return (
        <>
        {adData && <Banner {...adData}/>}
        
        </>
     );
};
 
export default AdvertisementBanner;