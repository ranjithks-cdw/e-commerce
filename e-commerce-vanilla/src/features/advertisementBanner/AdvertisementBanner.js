import Banner from "./banner/Banner.js";

const AdvertisementBanner = {
    addComponent: () => addComponent(),
};

const addComponent = () => {
    const content = createDOM();

    return content;
};

const createDOM = () => {
    const data = getData();
    const banner = Banner.create({...data});

    return banner;
};

const getData = () => {
    const adData = {
        adName: "Title 1",
        adImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        adRedirectUrl: "/products"
    };
    return adData;
}

export default AdvertisementBanner;