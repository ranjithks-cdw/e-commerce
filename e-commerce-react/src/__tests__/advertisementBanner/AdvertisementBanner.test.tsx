import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import AdvertisementBanner from "../../features/advertisementBanner/AdvertisementBanner"

describe('Advertisement Banner', () => {
    it("renders component", async () => {
        const {container} = render(<AdvertisementBanner />);
        expect(container).not.toBeNull();
    })

    it("matches snapshot", () => {
        const container = renderer.create(<AdvertisementBanner />).toJSON;
        expect(container).toMatchSnapshot();
    })
})