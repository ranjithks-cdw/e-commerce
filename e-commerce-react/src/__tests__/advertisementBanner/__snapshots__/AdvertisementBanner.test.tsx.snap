// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advertisement Banner matches snapshot 1`] = `[Function]`;
